projectsRouter.js

```diff
@@ -0,0 +1,197 @@
+
+//     projects.getProjectActions(req.params.id)
+//     .then(actions => {
+//         if(actions) {
+//             res.status(200).json(actions)
+//         } else {
+//             res.status(404).json({ message: "The project does not have actions." });
+//         }
+//     })
+//     .catch(err => {
+//         next(err)
+//     })
+
+// });
+
+router.get("/:id/actions", validateProjectId(), async (req, res, next) => {
+  try {
+    const actions = await projects.getProjectActions(req.params.id);
+    if (actions) {
+      res.status(200).json(actions);
+    } else {
+      res.status(404).json({ message: "The project does not have actions." });
+    }
+  } catch (error) {
+    next(error);
+  }
+});
+
+//POST a new project
+
+// router.post("/", validateProject(), (req, res, next) => {
+//     projects
+//       .insert(req.body)
+//       .then(project => {
+//         res.status(201).json(project);
+//       })
+//       .catch(err => {
+//         next(err);
+//       });
+//   });
+
+router.post("/", validateProject(), async (req, res, next) => {
+  try {
+    const project = await projects.insert(req.body);
+    res.status(201).json(project);
+  } catch (error) {
+    next(error);
+  }
+});
+
+//DELETE project
+
+//  router.delete("/:id", validateProjectId(), (req, res, next) => {
+//     projects.remove(req.params.id)
+//     .then(count => {
+//       if (count > 0) {
+//         res.status(200).json({ message: 'The project has been deleted' });
+//       } else {
+//         res.status(404).json({ message: 'The project could not be found' });
+//       }
+//     })
+//     .catch(error => {
+//       next(error)
+//     });
+
+//   });
+
+router.delete("/:id", validateProjectId(), async (req, res, next) => {
+  try {
+    const count = await projects.remove(req.params.id);
+    if (count > 0) {
+      res.status(200).json({ message: "The project has been deleted" });
+    } else {
+      res.status(404).json({ message: "The project could not be found" });
+    }
+  } catch (error) {
+    next(error);
+  }
+});
+
+//PUT
+
+// router.put("/:id", validateProjectId(), validateProject(), (req, res, next) => {
+//   projects
+//     .update(req.params.id, req.body)
+//     .then(project => {
+//       res.status(200).json(project);
+//     })
+//     .catch(error => {
+//       next(error);
+//     });
+// });
+
+router.put(
+  "/:id",
+  validateProjectId(),
+  validateProject(),
+  async (req, res, next) => {
+    try {
+      const project = await projects.update(req.params.id, req.body);
+      res.status(200).json(project);
+    } catch (error) {
+      next(error);
+    }
+  }
+);
+
+//Custom Middleware
+
+//checks project's body (name and description) for POST and PUT request (not for api/projects/:id/actions)
+function validateProject() {
+  return (req, res, next) => {
+    if (!req.body) {
+      return res.status(400).json({ message: "missing project data" });
+    } else if (!req.body.name && !req.body.description) {
```

Comment:
The use of `&&` here can cause an unexpected bug.
If just one of the fields is present, this will bypass the validation and fail on the database.